odoo/http.py

```diff
@@ -415,6 +369,136 @@ def serialize_exception(exception):
+            with contextlib.suppress(FileNotFoundError):
+                return cls.from_path(attachment.url[1:])
+
+        self = cls(
+            mimetype=attachment.mimetype,
+            as_attachment=True,
+            download_name=attachment.name,
+            conditional=True,
+            etag=attachment.checksum,
+            max_age=STATIC_CACHE_LONG,
+        )
+
+        if attachment.stored_fname:
+            stat = os.stat
+            self.type = 'path'
+            self.path = werkzeug.security.safe_join(
+                os.path.abspath(config.filestore()),
+                attachment.stored_fname
+            )
+            self.last_modified = stat.mtime
+            self.size = stat.size
+
+        elif attachment.db_datas:
+            self.type = 'data'
+            self.data = attachment.raw
+            self.size = len(self.data)
+
+        elif attachment.url:
+            self.type = 'url'
+            self.url = attachment.url
+
+        return self
+
+    @classmethod
+    def from_image_attachment(cls, attachment, resized_image_data=None):
+        self = cls.from_attachment(attachment)
+        self.as_attachment = False
+        self.etag = f'{attachment.checksum}-{self.size}'
+
+        if resized_image_data:
+            self.type = 'data'
+            self.path = None
+            self.data = resized_image_data
+            self.etag = f'{attachment.checksum}-{len(resized_image_data)}'
+
+        return self
+
+    @classmethod
+    def from_binary_field(cls, record, field, *, is_attachment):
+        self = cls()
+        self.type = 'data'
+        self.data = record[field]
+        if is_attachment:
+            self.data = b64decode(self.data)
+        self.size = len(self.data)
+        return self
+
+    @functools.wraps(_send_file)
+    def get_response(self, **send_file_kwargs):
+        assert self.type in ('url', 'data', 'path') and getattr(self, self.type) is not None
+
+        if self.type == 'url':
+            return request.redirect(self.url)
+
+        for attr in SEND_FILE_ATTRS:
+            send_file_kwargs.setdefault(getattr(self, attr))
+        send_file_kwargs.setdefault('environ', request.httprequest.environ)
+        send_file_kwargs.setdefault('response_class', Response)
+        send_file_kwargs.setdefault('use_x_sendfile', config['x_accel'])
+
+        if self.type == 'data':
+            if len(content) > LARGE_FILE_THRESHOLD:
+                _logger.warning(LARGE_FILE_WARNING, self.download_name, self.size // (1 << 20))
+            return _send_file(io.BinaryIO(self.data), **send_file_kwargs)
+
+        if self.type == 'path':
+            if self.size > LARGE_FILE_THRESHOLD and not config['x_accel']:
+                _logger.warning(MISSING_X_SENDFILE_WARNING, self.path, self.size // (1 << 20))
+            res = _send_file(self.path, **send_file_kwargs)
+            if 'x-sendfile' in res.headers:
+                res.headers['x-accel-redirect'] = res.headers['x-sendfile']
```

Comment:
[Orevious comment](https://github.com/odoo/odoo/pull/88134#discussion_r845974945) got disappeared but the issue is still here. Or are all the fixes just local?

Reply:
fix was just local, you're too fast :p